src/main/java/model/Student.java

```diff
@@ -1,15 +1,15 @@
 package model;
 
-import jakarta.persistence.Entity;
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.Id;
+import jakarta.persistence.*;
 
 import java.util.Objects;
 @Entity
 public class Student {
     @Id
     @GeneratedValue
     private long id;
+    @OneToMany( mappedBy= "studentCollection")
```

Comment:
```suggestion
    @ManyToOne
    @JoinColumn(name = "faculty_id")
```
Тут тоже меняем связь